dotsql.go

```diff
@@ -165,8 +179,17 @@ func Load(r io.Reader) (*DotSql, error) {
 	scanner := &Scanner{}
 	queries := scanner.Run(bufio.NewScanner(r))
 
+	templates := make(map[string]*template.Template, len(queries))
+	for k, v := range queries {
+		var err error
+		templates[k], err = template.New(k).Parse(v)
```

Comment:
So if there's an error here, we'll create an entry on `templates` with a nil value, this will have a direct impact on:
```
func (d Dotsql) lookup(...) {

if template == nil {
		return "", nil
	}

}
```

My intuition is that it's better to not create an entry here when an error is returned, ie:

```
	tpl, err := template.New(k).Parse(v)
	if err != nil {
		return nil, fmt.Errorf("could not parse query template: %w", err)
	}

	templates[k] = tpl
```

so that we do not create extra code on a different function that interfere with clarity